extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java

```diff
@@ -117,11 +122,28 @@ public KubernetesPortBuildItem kubernetes() {
     VertxWebRouterBuildItem initializeRouter(VertxHttpRecorder recorder,
             CoreVertxBuildItem vertx,
             List<RouteBuildItem> routes,
+            NonApplicationRootPathBuildItem nonApplicationRootPath,
+            BuildProducer<VertxNonApplicationRouterBuildItem> frameworkRouterBuildProducer,
             ShutdownContextBuildItem shutdown) {
 
         RuntimeValue<Router> router = recorder.initializeRouter(vertx.getVertx());
+        RuntimeValue<Router> frameworkRouter = recorder.initializeRouter(vertx.getVertx());
+        boolean frameworkRouterFound = false;
+
         for (RouteBuildItem route : routes) {
-            recorder.addRoute(router, route.getRouteFunction(), route.getHandler(), route.getType());
+            if (nonApplicationRootPath.isSeparateRoot() && route.isFrameworkRoute()) {
+                frameworkRouterFound = true;
+                recorder.addRoute(frameworkRouter, route.getRouteFunction(), route.getHandler(), route.getType());
+            } else {
+                recorder.addRoute(router, route.getRouteFunction(), route.getHandler(), route.getType());
+            }
+        }
+
+        if (frameworkRouterFound && nonApplicationRootPath.isSeparateRoot()) {
+            // Handle redirects from old paths to new non application endpoint root
+            recorder.addNonApplicationPathRedirect(router, frameworkRouter, nonApplicationRootPath.getFrameworkRootPath());
+
```

Comment:
Should this be disable-able?

Reply:
I think httpBuildTimeConfig.redirectToNonApplicationRootPath came in while I was reviewing .. ;)